source/Module/STIG/Functions.Checklist.ps1

```diff
@@ -240,18 +240,36 @@ function New-StigCheckList
     $writer = [System.Xml.XmlWriter]::Create($OutputPath.FullName, $xmlWriterSettings)
 
     $writer.WriteStartElement('CHECKLIST')
-
-    #region ASSET
-
     $writer.WriteStartElement("ASSET")
 
+    try
+    {
+        if ($hostName -eq $env:COMPUTERNAME)
+        {
+            [string]$localMac = (Get-NetAdapter | Where-Object {$_.Status -eq "Up"} | Select-Object MacAddress | Select-Object -First 1).MacAddress
+            [string]$localIP = (Get-NetIPAddress -AddressFamily IPV4 | Where-Object { $_.IpAddress -notlike "127.*" } | Select-Object -First 1).IPAddress
+            [string]$localFQDN = [System.Net.DNS]::GetHostByName($env:ComputerName).HostName
+        }
+        else
+        {
+            [string]$localMac = Invoke-Command $hostName -Scriptblock {return (Get-NetAdapter | Where-Object {$_.Status -eq "Up"} | Select-Object MacAddress | Select-Object -First 1).MacAddress}
```

Comment:
Named params over positional.  As well as stacking Select-Object, should be cleaned up either via grouping and using a property reference or creating an additional variable like the comment above.

Reply:
Added Named Parameters